### PR TITLE
Mustachio: Add getProperty(), and end-to-end Renderer test

### PR DESCRIPTION
**Pull request by srawlins**
The generated code now always references the library that generated it, on the
assumption that it needs types from that file. This is not necessarily correct,
but is a decent approximation until we start using code_builder.

RendererBase now needs to know about the renderer of the parent context type,
so it gets a new field.

Property and each renderer's propertyMap gets corrected code; the runtime tests
revealed some runtime type bugs.

### CODE REVIEW

**Comment by srawlins** [comment]
@jcollins-g what do you think about the try-publish failure? Moving the builder to lib/ would necessitate (?) that `package:build` move to be a proper dep. Yuck. Is this ignorable?

**Comment by srawlins** [comment]
@jcollins-g OK I think I've got this working; I've added a second build target in build.yaml which builds testing code. build_runner refused to generate any files in `testing/` (I still don't get that), so my test file is at `test/mustachio/foo.dart`.

File: test/mustachio/foo.dart [hunk 1]
```diff
@@ -0,0 +1,9 @@
+@Renderer(#renderFoo, Context<Foo>())
```
Comment (jcollins-g) [inline test/mustachio/foo.dart:1]: Since this is checked in it probably belongs in the grinder's rebuild check?

File: build.yaml [hunk 2]
```diff
@@ -16,8 +16,26 @@ targets:
     sources:
       - tool/builder.dart
       - tool/mustachio/builder.dart
+  testing:
```
Comment (jcollins-g) [inline build.yaml:19]: Does this change how the builder is invoked to generate foo.dart?

File: tool/mustachio/codegen_runtime_renderer.dart [hunk 3]
```diff
@@ -191,7 +220,8 @@ class ${renderer._rendererClassName}${renderer._typeParametersString}
 
     if (property.isPrivate || property.isStatic || property.isSetter) return;
     _buffer.writeln("'${property.name}': Property(");
-    _buffer.writeln('getValue: ($typeName c) => c.${property.name},');
+    _buffer
+        .writeln('getValue: (Object c) => (c as $typeName).${property.name},');
```
Comment (jcollins-g) [inline tool/mustachio/codegen_runtime_renderer.dart:224]: I don't object, but I am curious what the motivation is for this change?

**Review by jcollins-g — APPROVED** [review]
Approved assuming that a rebuilding check for foo.dart is added.

File: build.yaml (hunk 2 above)
Comment (srawlins) [reply build.yaml:19]: No, `pub run build_runner build` still works fine.

File: tool/mustachio/codegen_runtime_renderer.dart (hunk 3 above)
Comment (srawlins) [reply tool/mustachio/codegen_runtime_renderer.dart:224]: The code as it was was a runtime error. The `Map<String, Property>` must contain `Property<Object>` objects, and the functions of a `Property<Object>` object must take an `Object` as a parameter. A bummer, and potential performance hit, but necessary without some ingenuity. :(

File: test/mustachio/foo.dart (hunk 1 above)
Comment (srawlins) [reply test/mustachio/foo.dart:1]: Done!

**Pull request merged**